rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy

```diff
@@ -577,11 +593,32 @@ class MenuController extends ControllerBase implements ApplicationContextAware{
                 formatted
         )
     }
+
+    private def getConfiguredMaxPerPage(int defaultMax) {
+        if(guiPaginationEnabled) {
+            return grailsApplication.config.rundeck.gui.paginatejobs.max.per.page.isEmpty() ? defaultMax : grailsApplication.config.rundeck.gui.paginatejobs.max.per.page.toInteger()
+        }
+        return null
+    }
+
+    private boolean getGuiPaginationEnabled() {
+        grailsApplication.config.rundeck.gui.paginatejobs.enabled in ["true",true]
+    }
+
+    private boolean getApiPaginationEnabled() {
+        if(grailsApplication.config.rundeck.api.paginatejobs.enabled != null)
+            return grailsApplication.config.rundeck.api.paginatejobs.enabled in ["false",false]
+        
+        return true
+    }
+
     private def listWorkflows(ScheduledExecutionQuery query,AuthContext authContext,String user) {
         long start=System.currentTimeMillis()
         if(null!=query){
             query.configureFilter()
         }
+
+        query.max = getConfiguredMaxPerPage(10)
```

Comment:
I think this should be removed.  Wouldn't this override the max provided in the query object, even for api requests?